src/primitives/block.h

```diff
@@ -110,6 +110,17 @@ class CBlock : public CBlockHeader
         return block;
     }
 
+	bool LookupTransactionIndex(uint256 txid, uint32_t& index) 
```

Comment:
In 1d376496fe9ac7110d0fac773ae5b5d9ef32aa59:

I'd return a `std::optional<uint32_t>` rather than returning a `bool` and using a `uint32_t` outparam.

Reply:
I see very few uses of the std::optional in the rest of the code base, is there a reason for this?

Reply:
Hmm never mind I must have done a selective git grep, Thanks!